tgui/packages/tgui/interfaces/AppearanceChanger.js

```diff
@@ -0,0 +1,310 @@
+          )}
+          {!!has_colours && (
+            <LabeledList.Item label="Colors">
+              {!!change_eye_color && (
+                <Button
+                  content="Change eye color"
+                  onClick={
+                    () => act('eye_color')
+                  }
+                />
+              )}
+              {!!change_skin_tone && (
+                <Button
+                  content="Change skin tone"
+                  onClick={
+                    () => act('skin_tone')
+                  }
+                />
+              )}
+              {!!change_skin_color && (
+                <Button
+                  content="Change skin color"
+                  onClick={
+                    () => act('skin_color')
+                  }
+                />
+              )}
+              {!!change_head_accessory_color && (
+                <Button
+                  content="Change head accessory color"
+                  onClick={
+                    () => act('head_accessory_color')
+                  }
+                />
+              )}
+              {!!change_hair_color && (
+                <Button
+                  content="Change hair color"
+                  onClick={
+                    () => act('hair_color')
+                  }
+                />
+              )}
+              {!!change_secondary_hair_color && (
+                <Button
+                  content="Change secondary hair color"
+                  onClick={
+                    () => act('secondary_hair_color')
+                  }
+                />
+              )}
+              {!!change_facial_hair_color && (
+                <Button
+                  content="Change facial hair color"
+                  onClick={
+                    () => act('facial_hair_color')
+                  }
+                />
+              )}
+              {!!change_secondary_facial_hair_color && (
+                <Button
+                  content="Change secondary facial hair color"
+                  onClick={
+                    () => act('secondary_facial_hair_color')
+                  }
+                />
+              )}
+              {!!change_head_marking_color && (
+                <Button
+                  content="Change head marking color"
+                  onClick={
+                    () => act('head_marking_color')
+                  }
+                />
+              )}
+              {!!change_body_marking_color && (
+                <Button
+                  content="Change body marking color"
+                  onClick={
+                    () => act('body_marking_color')
+                  }
+                />
+              )}
+              {!!change_tail_marking_color && (
+                <Button
+                  content="Change tail marking color"
+                  onClick={
+                    () => act('tail_marking_color')
+                  }
+                />
+              )}
```

Comment:
I'd probably simplify that with a list of Objects containing the name of the key in `data` to check, the label name and the action.

Reply:
I cant think of a pheasible way to do this, but youre welcome to improve it later